### ISSUE TITLE
NPE exception when attempting to validate against a profile that does not contain a snapshot

### ISSUE THREAD

**Issue by michaelwschroeder** [IBM/FHIR#1230]
When attempting to validate against a profile that does not contain a snapshot, we get a fatal error from FHIR due to a null pointer exception in the FHIR code. Would be good to get a meaningful error message. 

stacktrace:
```
java.lang.NullPointerException
	at com.ibm.fhir.profile.ProfileSupport.getKeys(ProfileSupport.java:241)
	at com.ibm.fhir.profile.ProfileSupport.computeConstraints(ProfileSupport.java:66)
	at com.ibm.fhir.profile.ProfileSupport.lambda$getConstraints$1(ProfileSupport.java:216)
	at com.ibm.fhir.profile.ProfileSupport$$Lambda$147/0000000000000000.apply(Unknown Source)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1671)
	at com.ibm.fhir.profile.ProfileSupport.getConstraints(ProfileSupport.java:216)
	at com.ibm.fhir.profile.ProfileSupport.getConstraints(ProfileSupport.java:185)
	at com.ibm.fhir.validation.FHIRValidator$ValidatingNodeVisitor.validate(FHIRValidator.java:300)
	at com.ibm.fhir.validation.FHIRValidator$ValidatingNodeVisitor.doVisit(FHIRValidator.java:259)
	at com.ibm.fhir.path.visitor.FHIRPathDefaultNodeVisitor.visit(FHIRPathDefaultNodeVisitor.java:81)
	at com.ibm.fhir.path.FHIRPathResourceNode.accept(FHIRPathResourceNode.java:176)
	at com.ibm.fhir.validation.FHIRValidator$ValidatingNodeVisitor.validate(FHIRValidator.java:244)
	at com.ibm.fhir.validation.FHIRValidator$ValidatingNodeVisitor.access$100(FHIRValidator.java:230)
	at com.ibm.fhir.validation.FHIRValidator.validate(FHIRValidator.java:172)
	... 66 more 
```

**Comment by prb112** [comment]
PR https://github.com/IBM/FHIR/pull/1235

**Comment by prb112** [comment]
Fix is in 4.3.0
